Assets/BossRoom/Scripts/Server/Game/Character/ServerCharacterMovement.cs

```diff
@@ -128,6 +128,26 @@ public void CancelMove()
             m_MovementState = MovementState.Idle;
         }
 
+        /// <summary>
+        /// Instantly moves the character to a new position. NOTE: this cancels any active movement operation!
+        /// </summary>
+        /// <param name="newPosition">new coordinates the character should be at</param>
+        public void Teleport(Vector3 newPosition)
+        {
+            CancelMove();
+            if (!m_NavMeshAgent.Warp(newPosition))
+            {
+                // warping failed! We're off the navmesh somehow. Weird... but we can still teleport
+                Debug.LogWarning($"NavMeshAgent.Warp({newPosition}) failed!", gameObject);
+                transform.position = newPosition;
+            }
+
+            m_Rigidbody.position = transform.position;
+            m_Rigidbody.rotation = transform.rotation;
+
+            m_NetworkCharacterState.RecvTeleportClientRpc(newPosition);
```

Comment:
I'm curious, why isn't this part of the recvDoClientRPC in the action? Why do we need a separate RPC for the teleport?
Why is the action client RPC not just action anticipation?

Reply:
So in other words, why two RPC calls for one action?

Reply:
(Sorry, I didn't realize this comment had been marked as Resolved before I started working on it -- but I think it's worth addressing anyway)

Since the point of this RPC was the teleport, I started with a clean version of teleportation and then made an Action that shows how to teleport with a unique visualization. This way it's obvious how to add something like a teleport platform - just call `ServerCharacterMovement.Teleport()`. 

But it's true the visualization doesn't need the extra RPC. I've changed `ServerCharacterMovement.Teleport()` to have a second parameter indicating whether the RPC is needed, and set it to false here.

I've also removed the code that passed the teleportation event to the ActionFX as it isn't necessary.